assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/AudienceTiles.js

```diff
@@ -338,189 +523,171 @@ export default function AudienceTiles( { Widget, widgetLoading } ) {
-						);
-					} )?.metricValues;
+						const pageviews =
+							Number( currentMetricValues?.[ 3 ]?.value ) || 0;
+						const prevPageviews =
+							Number( previousMetricValues?.[ 3 ]?.value ) || 0;
 
-					const audienceName =
-						audiences?.filter(
-							( { name } ) => name === audienceResourceName
-						)?.[ 0 ]?.displayName || '';
+						const topCities = topCitiesReport?.[ index ];
 
-					const audienceSlug =
-						audiences?.filter(
-							( { name } ) => name === audienceResourceName
-						)?.[ 0 ]?.audienceSlug || '';
+						const topContent = topContentReport?.[ index ];
 
-					const visitors =
-						Number( currentMetricValues?.[ 0 ]?.value ) || 0;
-					const prevVisitors =
-						Number( previousMetricValues?.[ 0 ]?.value ) || 0;
+						const topContentTitles = {};
 
-					const visitsPerVisitors =
-						Number( currentMetricValues?.[ 1 ]?.value ) || 0;
-					const prevVisitsPerVisitors =
-						Number( previousMetricValues?.[ 1 ]?.value ) || 0;
-
-					const pagesPerVisit =
-						Number( currentMetricValues?.[ 2 ]?.value ) || 0;
-					const prevPagesPerVisit =
-						Number( previousMetricValues?.[ 2 ]?.value ) || 0;
-
-					const pageviews =
-						Number( currentMetricValues?.[ 3 ]?.value ) || 0;
-					const prevPageviews =
-						Number( previousMetricValues?.[ 3 ]?.value ) || 0;
-
-					const topCities = topCitiesReport?.[ index ];
-
-					const topContent = topContentReport?.[ index ];
+						topContentPageTitlesReport?.[ index ]?.rows?.forEach(
+							( row ) => {
+								topContentTitles[
+									row.dimensionValues[ 0 ].value
+								] = row.dimensionValues[ 1 ].value;
+							}
+						);
 
-					const topContentTitles = {};
+						const isPartialData =
+							partialDataStates[ audienceResourceName ];
+						const isZeroData = hasZeroDataForAudience(
+							report,
+							audienceResourceName
+						);
 
-					topContentPageTitlesReport?.[ index ]?.rows?.forEach(
-						( row ) => {
-							topContentTitles[ row.dimensionValues[ 0 ].value ] =
-								row.dimensionValues[ 1 ].value;
+						// Return loading tile if data is not yet loaded.
+						if (
+							loading ||
+							isZeroData === undefined ||
+							isPartialData === undefined
+						) {
+							return (
+								<Widget key={ audienceResourceName } noPadding>
+									<AudienceTileLoading />
+								</Widget>
+							);
 						}
-					);
-
-					const isPartialData =
-						partialDataStates[ audienceResourceName ];
-					const isZeroData = hasZeroDataForAudience(
-						report,
-						audienceResourceName
-					);
-
-					return (
-						<AudienceTile
-							loaded={ ! loading }
-							key={ audienceResourceName }
-							title={ audienceName }
-							infoTooltip={
-								<AudienceTooltipMessage
-									audienceName={ audienceName }
-									audienceSlug={ audienceSlug }
+
+						// If errored, skip rendering.
+						if (
+							individualTileErrors[ audienceResourceName ]
+								?.length > 0
```

Comment:
This array should always be defined at this point, so we shouldn't need the conditional chaining operator:

```suggestion
							individualTileErrors[ audienceResourceName ]
								.length > 0
```